include/flatbuffers/flatbuffers.h

```diff
@@ -395,6 +395,61 @@ template<typename T> static inline size_t VectorLength(const Vector<T> *v) {
   return v ? v->size() : 0;
 }
 
+// Lexicographically compare two strings (possibly containing nulls), and
+// return true if the first is less than the second.
+static inline bool StringLessThan(const char *a_data, uoffset_t a_size,
+                                  const char *b_data, uoffset_t b_size) {
+  const auto cmp = memcmp(a_data, b_data, (std::min)(a_size, b_size));
+  return cmp == 0 ? a_size < b_size : cmp < 0;
+}
```

Comment:
Is this method needed?
One can use std::lexicographical_compare with or without a comparator (ignore case for example).

Reply:
This function was not introduced by me, I had to shuffle the code around so git diff shows it unfortunately.